### ISSUE TITLE
minikube issue when not providing an IP address for ssh driver

### ISSUE THREAD

**Issue by tejas0026** [kubernetes/minikube#10266]
* Starting control plane node minikube in cluster minikube
! StartHost failed, but will try again: config: please provide an IP address
* Failed to start ssh bare metal machine. Running "minikube delete" may fix it:                                                                                                              config: please provide an IP address

X Exiting due to GUEST_PROVISION: Failed to start host: config: please provide an IP address
*
* If the above advice does not help, please let us know:
  - https://github.com/kubernetes/minikube/issues/new/choose

**Comment by afbjorklund** [comment]
Hmm, this is the same question as in #10244 - wonder if the documentation needs updating ?

https://minikube.sigs.k8s.io/docs/drivers/ssh/

**Comment by afbjorklund** [comment]
Added PR #10269 for this issue

`X Exiting due to DRV_NOT_DETECTED: No possible driver was detected. Try specifying --driver, or see https://minikube.sigs.k8s.io/docs/start/`

**Comment by mdtnpll** [comment]
Open a PowerShell console as Administrator, and run the following command:

Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All

**Comment by afbjorklund** [comment]
Fixed in minikube 1.17.1.

Now it will not pick the "ssh" driver by default, but instead try to find an available driver.

```
* Unable to pick a default driver. Here is what was considered, in preference order:
  - vmware: Not installed: exec: "docker-machine-driver-vmware": executable file not found in $PATH
  - docker: Not installed: exec: "docker": executable file not found in $PATH
  - kvm2: Not installed: exec: "virsh": executable file not found in $PATH
  - none: Not installed: exec: "iptables": executable file not found in $PATH
  - podman: Not installed: exec: "podman": executable file not found in $PATH
  - virtualbox: Not installed: unable to find VBoxManage in $PATH

X Exiting due to DRV_NOT_DETECTED: No possible driver was detected. Try specifying --driver, or see https://minikube.sigs.k8s.io/docs/start/
```

If you _do_ want to use the new "ssh" driver, you need to select it using `--driver=ssh`

**Comment by Nagaraja-H** [comment]
1) install docker engine
2) run sudo usermod -aG docker $USER
3)  newgrp docker
3) minikube start --driver=docker

It worked for me

**Comment by RoFz** [comment]
I was also getting:

> I0505 15:25:48.157164   48952 main.go:130] libmachine: (minikube) DBG | executing: /Applications/VMware Fusion.app/Contents/Public/vmrun deleteVM /Users/rofz/.minikube/machines/minikube/minikube.vmx nogui
> W0505 15:25:48.306739   48952 out.go:241] 🤦  StartHost failed, but will try again: creating host: create: Error creating machine: Error in driver during machine creation: Machine didn't return an IP after 120 seconds, aborting

_/Applications/VMware\ Fusion.app/Contents/Public/vmrun start /Users/rofz/.minikube/machines/minikube/minikube.vmx nogui_ was saying only this:

> 2022-05-05T15:29:56.331| ServiceImpl_Opener: PID 50044
> Error: The operation was canceled

But when executed without the _nogui_ parameter (opening the minikube.vmx from the GUI also works), I've got this:

`Memory size 5433 MB is not a multiple of 4 MB.`
`Module 'MainMemEarly' power on failed. Failed to start the virtual machine`

After adding --memory=6144 (or any other amount divisible by 4) the cluster starts fine.

Why does minikube choose an incompatible amount of RAM (5433) for its node(s)? I'm deploying a three-node cluster.

> I0505 15:23:35.835430   48952 start.go:126] createHost starting for "" (driver="vmware")
> I0505 15:23:35.883436   48952 out.go:203] 🔥  Creating vmware VM (CPUs=2, Memory=5433MB, Disk=20000MB) ...
> I0505 15:23:35.883590   48952 main.go:130] libmachine: Found binary path at /usr/local/bin/docker-machine-driver-vmware
> I0505 15:23:35.883622   48952 main.go:130] libmachine: Launching plugin server for driver vmware
> I0505 15:23:35.893550   48952 main.go:130] libmachine: Plugin server listening at address 127.0.0.1:57818
> I0505 15:23:35.893961   48952 main.go:130] libmachine: () Calling .GetVersion

**Comment by brandonros** [comment]
> Fixed in minikube 1.17.1.
> 
> Now it will not pick the "ssh" driver by default, but instead try to find an available driver.
> 
> ```
> * Unable to pick a default driver. Here is what was considered, in preference order:
>   - vmware: Not installed: exec: "docker-machine-driver-vmware": executable file not found in $PATH
>   - docker: Not installed: exec: "docker": executable file not found in $PATH
>   - kvm2: Not installed: exec: "virsh": executable file not found in $PATH
>   - none: Not installed: exec: "iptables": executable file not found in $PATH
>   - podman: Not installed: exec: "podman": executable file not found in $PATH
>   - virtualbox: Not installed: unable to find VBoxManage in $PATH
> 
> X Exiting due to DRV_NOT_DETECTED: No possible driver was detected. Try specifying --driver, or see https://minikube.sigs.k8s.io/docs/start/
> ```
> 
> If you _do_ want to use the new "ssh" driver, you need to select it using `--driver=ssh`

Unless I have some weird config file leftover from another time I tried this and forgot,

```
 $ minikube start
😄  minikube v1.28.0 on Darwin 12.6 (arm64)
🆕  Kubernetes 1.25.3 is now available. If you would like to upgrade, specify: --kubernetes-version=v1.25.3
✨  Using the ssh driver based on existing profile

❌  Exiting due to MK_USAGE: No IP address provided. Try specifying --ssh-ip-address, or see https://minikube.sigs.k8s.io/docs/drivers/ssh/
```

I don't think this is fixed automatically as of v1.28.0?
